django/db/models/sql/compiler.py

```diff
@@ -2060,9 +2106,12 @@ def pre_sql_setup(self):
         query.add_fields(fields)
         super().pre_sql_setup()
 
+        # If the table has a composite pk, idents are pre-selected because not all
+        # databases support expressions such as "(id_1, id_2) IN (SELECT ...)".
+        is_composite_pk = isinstance(meta.pk, CompositePrimaryKey)
```

Comment:
Can we have a database feature for the databases that *do* support better expressions here so we can avoid pre-selecting in those cases?

Reply:
Let me check!

Reply:
Yes, I managed to do this, please check.